src/server/game/AI/ScriptedAI/ScriptedCreature.cpp

```diff
@@ -547,6 +547,23 @@ void BossAI::UpdateAI(uint32 diff)
     DoMeleeAttackIfReady();
 }
 
+void BossAI::_DespawnAtEvade(uint32 delayToRespawn)
+{
+    uint32 corpseDelay = me->GetCorpseDelay();
+    uint32 respawnDelay = me->GetRespawnDelay();
+
+    me->SetCorpseDelay(1);
+    me->SetRespawnDelay(delayToRespawn);
```

Comment:
I prefer the `ASSERT`. Immediately tells the person writing the script "this doesn't work".